Marlin/src/module/stepper.cpp

```diff
@@ -2915,8 +2915,8 @@ hal_timer_t Stepper::block_phase_isr() {
           if (++index == IS_COMPENSATION_BUFFER_SIZE) index = 0;
         }
         FORCE_INLINE xy_long_t past_item(const uint16_t n) {
-          const uint16_t i = index + IS_COMPENSATION_BUFFER_SIZE - n;
-          return buffer[i < IS_COMPENSATION_BUFFER_SIZE ? i : i - IS_COMPENSATION_BUFFER_SIZE];
+          const int16_t i = int16_t(index) - n;
+          return buffer[i > 0 ? i : i + IS_COMPENSATION_BUFFER_SIZE];
```

Comment:
Being nitpicky here, but i think the int casting also costs an extra op, so this is still 3 and 4 ops

Reply:
The casting is intended to prevent the compiler complaining about signed underflow but the assembly language to subtract 16 bits from 16 bits should not differ. Perhaps a `static_cast` will help to prevent any runtime sign-ish behavior.

Reply:
In that case, your version is the cleanest